### DISCUSSION TITLE
Floating namespace syntax

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#5916] [Ideas]
# Feature
This allows you to write
```scala
mod NowThisIsReallyNice

pub def example(): Unit = ()
```
where `example` is in the `NowThisIsReallNice` module.

**Note**: This shorthand module can only happen if it's the first (non-comment) thing in the file, and it behaves as though the rest of the file is enclosed in `{...}`.

# Pros
* Decreases indentation (most files have a 4char left margin for little gain)
* Avoids annoying and confusing parsing errors when you forget the last `}`
* This is totally optional; no breaking changes
* [Consistency with Scala](https://docs.scala-lang.org/tour/packages-and-imports.html)
* [Consistent with Rust](https://doc.rust-lang.org/rust-by-example/mod/split.html)
# Cons
* You can't trivially see if something is in a namespace or top-level (but files/projects of any scrolling size should not use top-level IMO)
* This might make nested namespaces confusing, but we could consider disallowing this. In the below example, `f` is in namespace `A.B.f`
```scala
mod A
mod B {
    def f(): Unit = ()
}
```

**Comment by mlutze** [comment]
- Con: strange interaction with companion namespaces. Maybe we could have some sugar for this.

**Reply by JonathanStarup** [reply]
Yes, in the light of companions, I don't think it should have to be the first thing
```
enum MyEnum ...

mod MyEnum

...
```
